CONTRIBUTING.md

```diff
@@ -47,3 +47,73 @@ mvn -f sdk/{projectForlderDir}/pom.xml -Dgpg.skip clean install
+  <groupId>MyGroup</groupId>
+  <artifactId>MyArtifact</artifactId>
+  <version>1.0.0-preview.1</version> <!-- {x-version-update;MyGroup:MyArtifact;[current|dependency]} -->
+```
+The last element of the tag would be current or dependency depending on the criteria previously explained.
+
+In README files this ends up being slightly different. Because the version tag is inside an XML element that we've explicitly telling a user to copy/paste into their product the comment tag really didn't make sense here. Instead there are tags before and after the XML element tags which effectively says "there's a version somewhere in between these two tags, when you find the line that matches replace it with the appropriate version of the group:artifact defined in the tag."
+
+\[//]: # ({x-version-update-start;MyGroup:MyArtifact;dependency})
+\```xml
+  <groupId>MyGroup</groupId>
+  <artifactId>MyArtifact</artifactId>
+  <version>1.0.0-preview.1</version>
+\```
+\[//]: # ({x-version-update-end})
+
+
+What if I've got something that, for whatever reason, shoudln't be updated? There's a tag for that
+<!-- {x-version-exempt;<groupId>:<artifactId>;reason for excemption} -->
+In theory, absence of an x-version-update tag would do the same thing but the tooling is still being developed and eventually there will be checkin blockers if xml has a version element with no tag.
+
+
+### What does the process looks like?
+Let's say we've GA'd and I need to tick up the version of azure-storage libraries how would I do it? 
+1. I'd open up eng\versioning\version_client.txt and update the current-versions of the libraries are built and released as part of the azure storage pipeline. This list can be found in pom.service.xml under the sdk/storage directory. It's worth noting that any module entry starting with "../" isn't something that's released as part of the pipeline and once we GA these build dependencies for library components outside of a given area should go away and be replaced with downloading the appropriate dependency from Maven like we do for exxternal dependencies.
+2. Execute the update_versions python script
+python eng/versioning/update_versions.py --ut libary --bt client
+This will go through the entire source tree and update all of the references in the POM and README files with the updated versions. Git status will show all of the modified files.
+3. Review and submit a PR with the modified files.
+
+### Next steps: External dependencies, Management plane and service pipeline changes
+- External dependencies. Right now only there are only version files for client and data (eng\versioning\version_\[client|data\].txt) which only encompass the built binaries for their respective tracks. External dependencies for both client and data are next on the list which should allow modification of the parent/pom.xml to remove the list of version properties and dependency management sections which brings things one step closer to not having to publish the parent pom.
```

Comment:
"Right now only there are only" -> "Right now there are only"

Reply:
Updated. Will be updated with the next commit.